Library/ViewModels/ThanksViewModel.swift

```diff
@@ -199,6 +199,32 @@ public final class ThanksViewModel: ThanksViewModelType, ThanksViewModelInputs,
       reward: $0.reward,
       checkoutData: $0.checkoutData
     ) }
+
+    // FB CAPI
+    let graphUser = self.viewDidLoadProperty.signal.ignoreValues()
+      .switchMap { _ in
+        AppEnvironment.current.apiService
+          .fetchGraphUser(withStoredCards: false)
+          .ksr_delay(AppEnvironment.current.apiDelayInterval, on: AppEnvironment.current.scheduler)
+          .materialize()
+      }
+
+    _ = Signal.combineLatest(project, self.viewDidLoadProperty.signal.ignoreValues())
+      .combineLatest(with: graphUser)
+      .observeValues { projectSignal, graphUser in
+        let (project, _) = projectSignal
```

Comment:
Okay so getting the user data is a bit tricky. But we have conventions kind of in place to pass existing data back up the ladder of views in place of making a completely new network call that gets ALL the user data when really all we need is one field "email". 

So the way I would do this is start with `PledgePaymentMethodsViewModel`, because it already does the `fetchGraphUser` and has outputs with the names `notify` prefix to them. Create a new output `notifyFacebookCAPIUserData` and send the email from the existing `fetchGraphUser` call. The signal can be the same as the one you use to get the email for tracking the payment sheet appearance.

next step is the output in `PledgePaymentMethodsViewController`, which can call a delegate kind of like  
```
extension PledgeViewController: PledgePaymentMethodsViewControllerDelegate {
  func pledgePaymentMethodsViewController(
    _: PledgePaymentMethodsViewController,
    didSelectCreditCard paymentSource: PaymentSourceSelected
  ) {
    self.viewModel.inputs.creditCardSelected(with: paymentSource)
  }
```
that stores the user data (just the email) within the `PledgeViewModel` as maybe a mutable property?. Now we see that `ThanksPageData` is created in the `PledgeViewModel` before triggering the thanks page. So pass along the stored value alongside existing checkout data.

Reply:
Ok I was think I was able to follow this pattern. IMO though this pattern is pretty confusing to follow, especially as someone still pretty new to ReactiveSwift. It takes more effort just to traced down where that signal is coming from and I'd prefer to have a single source of truth for pulling this information. However, I see how making a new network call for every time we want a user's email isn't great.